src/lib/trace/lttng/provider_circuit.h

```diff
@@ -0,0 +1,150 @@
+/* Copyright (c) 2020, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file provider_circuit.h
+ * \brief Header file that contains the LTTng probes for the tracepoint of the
+ *        circuit subsystem.
+ **/
+
+#undef TRACEPOINT_PROVIDER
+#define TRACEPOINT_PROVIDER tor_circuit
+
+#undef TRACEPOINT_INCLUDE
+#define TRACEPOINT_INCLUDE "./lib/trace/lttng/provider_circuit.h"
+
+#if !defined(_TRACEPOINT_TOR_CIRCUIT_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
+#define _TRACEPOINT_TOR_CIRCUIT_H
+
+#include <lttng/tracepoint.h>
+
+#include "core/or/circuitlist.h"
+#include "core/or/or.h"
+#include "core/or/origin_circuit_st.h"
+
+TRACEPOINT_ENUM(tor_circuit, purpose,
```

Comment:
For every single possible event we should have comments explaining what it means.

Reply:
Agree